lib/active_record_proxy_adapters/primary_replica_proxy.rb

```diff
@@ -16,7 +16,7 @@ class PrimaryReplicaProxy # rubocop:disable Metrics/ClassLength
       /\A\s*select.+(nextval|currval|lastval|get_lock|release_lock|pg_advisory_lock|pg_advisory_unlock)\(/i
     ].map(&:freeze).freeze
     # All queries that match these patterns should be sent to the replica database
-    SQL_REPLICA_MATCHERS     = [/\A\s*(select|with.+\)\s*select)\s/i].map(&:freeze).freeze
+    SQL_REPLICA_MATCHERS     = [/\A\s*(select|with[\s\S]*\)\s*select)\s/i].map(&:freeze).freeze
```

Comment:
Hello @OskarEichler. Thanks for the contribution!

We need at least one whitespace caracter after  `with` though. Could you please modify it to this?

```suggestion
    SQL_REPLICA_MATCHERS     = [/\A\s*(select|with\s[\s\S]*\)\s*select)\s/i].map(&:freeze).freeze
```

Your current proposal matches this string:

```sql
WITHblog_ids AS (SELECT blogs.id)
SELECT * FROM posts where posts.blog_id = blogs.id;
```
Without the change: https://rubular.com/r/J1PIogAKRn5cGr
With the change: https://rubular.com/r/32xQUIIb8rbP8y

Reply:
Good call!